src/types.rs

```diff
@@ -64,3 +64,22 @@ impl fmt::Display for Diagnostic {
     )
   }
 }
+
+#[derive(Debug)]
+pub struct Diagnostics(pub Vec<Diagnostic>);
```

Comment:
question: should this not be under a feature flag?

Reply:
I suspect it doesn't get flagged as not used because it is `pub`, does it need to be `pub` or is `pub (crate)` sufficient?

Reply:
I'll rename this to `DiagnosticsError`. I think it would be useful outside this crate in some cases when you want to have an error of a collection of diagnostics.